CA.md

```diff
@@ -0,0 +1,73 @@
+# Example CA Usage
+
+Although there is more than one way to handle extensions (grants/identities) and
+signing certificates, HIBA includes a `hiba-ca.sh` tool and `hiba-gen` binary.
+The following serves as a simple example of a way to get started with HIBA.
+
+Once you have completed the
+[compilation instructions](https://github.com/google/hiba/blob/main/INSTALL.md#compilation)
+you will have a `hiba-gen` and `hiba-ca.sh` in your hiba directory. You can now
+create CA keys, grants, identities, grant eligibility, and sign host and user
+certificates with these extensions. The following assumes you are in your hiba
+directory. The CA structure will be in `~/.hiba-ca` by default.
+
+1. Generate a CA keypair.
+    ```
+    $ ./hiba-ca.sh -c -- -N ""
+    == Generating CA keys in ~/.hiba-ca/g/ca
+    == Done ==
+    ```
+1. Generate a test user keypair (alternatively you can copy existing keys to
+`~/.hiba-ca/users/`).
+    ```
+    $ ./hiba-ca.sh -c -u -I testuser -- -N ""
+    == Generating user keys in ~/.hiba-ca/g/users/testuser
+    == Done ==
+    Identity created: Private: ~/.hiba-ca/users/testuser
+                      Public:  ~/.hiba-ca/users/testuser.pub
+    ```
+    * Alternatively you can copy existing keys with `./hiba-ca.sh -i -u -f
+      path_to_pubkey -I principal`
+1. Copy the OpenSSH server host public key from the OpenSSH server to your
+   machine and import it.
+    ```
+    ./hiba-ca.sh -i -h -f path_to_pubkey -I hostname
+    ```
+1. Create an identity for prod, with a domain of example.com.
+    ```
+    $ ./hiba-gen -i -f "~/.hiba-ca/policy/identities/prod" domain example.com
+    ```
+1. Create a grant called shell with a matching domain.
+    ```
+    $ ./hiba-gen -f "~/.hiba-ca/policy/grants/shell" domain example.com
+    ```
+1. Allow testuser to request this grant (shell).
+    ```
+    $ ./hiba-ca.sh -p -I testuser -H shell
+    == Done ==
+    User testuser is now eligible for [shell]
+    ```
+1. Sign the OpenSSH server public key into a certificate. -I is the hostname you
+   set when importing the host public key.
+   the public key in ~/.hiba-ca/hosts.
+    ```
+    $ ./hiba-ca.sh -s -h -I test -n test -H prod
```

Comment:
Not needed, the hostname is added as a principal already, this serves for adding extra principals (maybe the host IPv4 / IPv6 addresses).